qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -90,26 +90,22 @@ def set_calibration_options(self, **fields):
                 )
         self._calibration_options.update_options(**fields)
 
-    def run(
-        self,
-        backend: Backend,
-        analysis: bool = True,
-        experiment_data: Optional[ExperimentData] = None,
-        **run_options,
-    ) -> ExperimentData:
-        """Run an experiment, perform analysis, and update any calibrations.
+    def run_analysis(self, experiment_data, **options) -> ExperimentData:
```

Comment:
Moving the update here means that it will not be run until the job completes and the `post_processing_callback` is called:

https://github.com/Qiskit/qiskit-experiments/blob/c90320503fa7177110030af91ccbec1739848a1f/qiskit_experiments/framework/base_experiment.py#L143-L146

So then the `block_for_results` below is unnecessary. This makes the `BaseCalibrationExperiment` behavior more like `BaseExperiment`. However, as we have discussed, this behavior might be dangerously confusing because now you can run multiple calibrations in a row and the circuits for all of them will be generated at once (unless you call `block_for_results` on the experiment_data for each) and submitted using the original `Calibrations` state and then one by one they will update the `Calibrations` from their callbacks. Typically, one wants to wait for the results of the first calibration, update `Calibrations`, use the new `Calibrations` state to generate the circuits for the next calibration experiment, and repeat.